File: leetcode/leetcode_py/find-the-difference-of-two-arrays.py
'''
https://leetcode.com/problems/find-the-difference-of-two-arrays

1. Set
time: o(n)
space: o(n)
'''
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        s1 = set()
        for num in nums1:
            s1.add(num)
        s2 = set()
        for num in nums2:
            s2.add(num)

        res1 = []
        for num in s1:
            if num not in s2:
                res1.append(num)
            
        res2 = []
        for num in s2:
            if num not in s1:
                res2.append(num)
        return [res1, res2]

File: leetcode/leetcode_cpp/find-the-difference-of-two-arrays.cpp
/*
https://leetcode.com/problems/find-the-difference-of-two-arrays

1. Set
time: o(n)
space: o(n)
*/
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        vector<vector<int>> res(2);

        unordered_set<int> s1;
        for (int num : nums1) s1.insert(num);
        
        unordered_set<int> s2;
        for (int num : nums2) s2.insert(num);

        for (int num : s1) {
            if (!s2.count(num)) res[0].push_back(num);
        }
        for (int num : s2) {
            if (!s1.count(num)) res[1].push_back(num);
        }
        return res;
    }
};